### PR TITLE
[SAGE-316] Banner react variant

### PR DESCRIPTION
**Pull request by QuintonJason**
## Description

- [x] added react banner version

## Screenshots
Screen Shot 2022-05-03 at 7 03 21 PM

## Testing in `sage-lib`

Visit the banner react page and verify that the component is aligned with the rails version
- [React](http://localhost:4100/?path=/docs/sage-banner--default)
- [Rails](http://localhost:4000/pages/component/banner?tab=preview)

## Testing in `kajabi-products`

1. (**LOW**) Adds a new react component. Has no affect in the app

## Related

Closes [SAGE-316](https://kajabi.atlassian.net/browse/SAGE-316)

### CODE REVIEW

File: packages/sage-react/lib/themes/legacy/Banner/BannerContent.jsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  active: PropTypes.bool,
+  bannerContext: PropTypes.string,
+  children: PropTypes.node,
+  className: PropTypes.string,
+  dismissable: PropTypes.bool,
+  id: PropTypes.string,
+  link: PropTypes.string,
+  text: PropTypes.string,
+  type: PropTypes.oneOf(Object.values(BannerContent.TYPES)),
+};
```
Comment (ju-Skinner) [inline packages/sage-react/lib/themes/legacy/Banner/BannerContent.jsx:90]: We need to add specs for this as well.

**Review by ju-Skinner — CHANGES_REQUESTED** [review]
Addition of or Updates to spec coverage.

**Review by anechol — APPROVED** [review]
LGTM

**Review by ju-Skinner — APPROVED** [review]

**Review by philschanely — APPROVED** [review]
One quick tweak noted below but 👍 otherwise.

**Pull request merged**